### PR TITLE
Refactor useSendAccountBalances to avoid nesting a hook

### PR DESCRIPTION
**Pull request by youngkidwarrior**

### CODE REVIEW

**Comment by youngkidwarrior** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/150?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#150** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/150?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#149** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/149?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#148** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/148?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#147** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/147?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#146** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/146?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `dev`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @youngkidwarrior and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by 0xLe0x — COMMENTED** [review]
LGTM. I'm adding coingecko api for fetching token prices in another PR. they'll be used for calculating total balance.

File: packages/app/utils/useSendAccountBalances.ts [hunk 1]
```diff
@@ -2,35 +2,57 @@ import {
...
-      address: sendAccount?.address,
-      token: token === 'eth' ? undefined : (token as `0x${string}`),
-      query: { enabled: !!sendAccount },
-      chainId: baseMainnet.id,
-    })
-  }
+  //@todo this is improper use of a hook. Hooks should always be used at top level of component
+  const { data: tokenBalances, isPending: isPendingTokenBalances } = useReadContracts({
+    query: { enabled: !!sendAccount },
+    contracts: [
```
Comment (0xBigBoss) [inline packages/app/utils/useSendAccountBalances.ts:30]: nice. this is a much better way to do it. didn't know u could queue up multiple reads in the same hook with wagmi.

**Review by 0xBigBoss — APPROVED** [review]

File: packages/app/utils/useSendAccountBalances.ts (hunk 1 above)
Comment (0xLe0x) [reply packages/app/utils/useSendAccountBalances.ts:30]: ❤️

Comment (youngkidwarrior) [reply packages/app/utils/useSendAccountBalances.ts:30]: I should remove the todo

**Comment by youngkidwarrior** [comment]
### Merge activity

* **Mar 4, 10:46 AM EST**: @youngkidwarrior started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/0xsend/sendapp/150).

**Pull request merged**